### ISSUE TITLE
BUG: Immutable parser with flags

### ISSUE THREAD

**Issue by paradigmatic** [scopt/scopt#4]
When a try to define a simple immutable parser with a flag:

```
val parser =  new scopt.immutable.OptionParser[Int]("scopt", "2.x") { 
  def options = Seq(
    flag( "v", "verbose", "verbose mode"){ i => i+1 }
  )
}
```

I run into a compilation error:

```
[error]  found   : scopt.generic.FlagOptionDefinition[Int => Int]
[error]  required: scopt.generic.OptionDefinition[Int]
[error]     flag( "v", "verbose", "verbose mode"){ i => i+1 }
```

Can you fix the problem ?

**Comment by eed3si9n** [comment]
Apologies for the delayed response. I didn't notice that org owners don't automatically get issues notifications.
There's a pending pull req that should fix this case.

**Comment by eed3si9n** [comment]
published 2.1.0.

**Comment by paradigmatic** [comment]
Thanks a lot !
